cores/esp8266/PolledTimeout.h

```diff
@@ -47,16 +47,91 @@ struct YieldOrSkip
 
 } //YieldPolicy
```

Comment:
Since you're already playing with the code here, could I ask you to please add the following:
```cpp
template <unsigned long delayMs>
struct YieldAndDelay
{
  static void execute() {delay(delayMs);}
};
```

It essentially means "poll every delayMs" as opposed to the current "poll as fast as possible". The use case is to reduce the busy wait loop CPU time, which reduces power usage. I've seen the use pattern in some places, hence the need.

Example for building a type from this (untested):
```cpp
using periodicFast50Ms = esp8266::polledTimeout::timeoutTemplate<true, YieldPolicy::YieldAndDelay<50>, TimePolicy::TimeFastMillis>;
periodicFast50Ms timeout(500); //poll every 50ms, expire at 500ms
```